tls/s2n_resume.c

```diff
@@ -102,6 +104,74 @@ static int s2n_deserialize_resumption_state(struct s2n_connection *conn, struct
     return 0;
 }
 
+static int s2n_client_serialize_resumption_state(struct s2n_connection *conn, struct s2n_stuffer *to)
+{
+    /* Serialize session id */
+    GUARD(s2n_stuffer_write_uint8(to, S2N_STATE_WITH_SESSION_ID));
+    GUARD(s2n_stuffer_write_uint8(to, conn->session_id_len));
+    GUARD(s2n_stuffer_write_bytes(to, conn->session_id, conn->session_id_len));
+
+    /* Serialize session state */
+    GUARD(s2n_serialize_resumption_state(conn, to));
+
+    return 0;
+}
+
+static int s2n_client_deserialize_session_state(struct s2n_connection *conn, struct s2n_stuffer *from)
+{
+    uint8_t session_id_len;
+    GUARD(s2n_stuffer_read_uint8(from, &session_id_len));
+
+    if (session_id_len == 0 || session_id_len > S2N_TLS_SESSION_ID_MAX_LEN
+        || session_id_len > s2n_stuffer_data_available(from)) {
+        return -1;
```

Comment:
here and below - we should use S2N_ERROR instead and add appropriate error type for deserealization failures.